hphp/util/hash.h

```diff
@@ -59,23 +64,33 @@ inline size_t hash_int64_fallback(int64_t key) {
   return static_cast<size_t>(static_cast<uint32_t>(key));
 }
 
-ALWAYS_INLINE size_t hash_int64(int64_t k) {
-#if defined(USE_SSECRC) && defined(__SSE4_2__)
+inline size_t hash_int64(int64_t k) {
+#if defined(USE_HWCRC) && defined(__SSE4_2__)
   size_t h = 0;
   __asm("crc32q %1, %0\n" : "+r"(h) : "rm"(k));
   return h;
+#elif defined(USE_HWCRC) && defined(ENABLE_AARCH64_CRC)
+  size_t res;
+  __asm("crc32cx %w0, wzr, %x1\n" : "=r"(res) : "r"(k));
+  return res;
 #else
   return hash_int64_fallback(k);
 #endif
 }
 
+
 inline size_t hash_int64_pair(int64_t k1, int64_t k2) {
-#if defined(USE_SSECRC) && defined(__SSE4_2__)
+#if defined(USE_HWCRC) && defined(__SSE4_2__)
   // crc32 is commutative, so we need to perturb k1 so that (k1, k2) hashes
   // differently from (k2, k1).
   k1 += k1;
   __asm("crc32q %1, %0\n" : "+r" (k1) : "rm"(k2));
   return k1;
+#elif defined(USE_HWCRC) && defined(ENABLE_AARCH64_CRC)
+  size_t res;
+  k1 += k1;
+  __asm("crc32cx %w0, %w1, %x2\n" : "=r"(res) : "r"(k2), "r"(k1));
```

Comment:
I swapped k1 and k2 since k1 is now possibly 33 bits and the crc32cx instruction uses a 64-bit reg. for %x2.